### PR TITLE
[SWE-1192][BOOKING MODULE] Connect slider and navigation points to ea…

### PR DESCRIPTION
**Pull request by ArtemSuhoy**
…ch other

### CODE REVIEW

File: src/app/components/elements/NavigationDots/NavigationDots.styled.js [hunk 1]
```diff
@@ -14,12 +14,14 @@ const NavigationDotStyled = styled.li`
...
   margin-left: 8px;
   cursor: pointer;
-  &.active {
-    height: 8px;
-    width: 32px;
-    border-radius: 8px;
-    background-color: var(--booking-module-neutral);
-  }
+  ${(props) =>
+    props.active &&
```
[the remark this answers is not in the document]
Comment (5P4C3GR33D) [reply src/app/components/elements/NavigationDots/NavigationDots.styled.js:18]: Only if property is not from constants. To constants there's no need to add this check.

File: src/app/components/elements/NavigationDots/NavigationDots.js [hunk 2]
```diff
@@ -7,23 +7,31 @@ import PropTypes from 'prop-types'
...
 const NavigationDots = (props) => {
-  const { navigationConfig } = props
+  const {
+    screenSliderCurrentSlide,
+    handleInnerScreenSlide,
+    numberScreenSlides
+  } = props
+
+  //[COMPUTED_PROPERTIES]
+  const screenSlidesArray = [...Array(numberScreenSlides)]
```
Comment (olenashevchuk) [inline src/app/components/elements/NavigationDots/NavigationDots.js:17]: there is useless code

File: src/app/components/elements/NavigationDots/NavigationDots.js [hunk 3]
```diff
@@ -7,23 +7,31 @@ import PropTypes from 'prop-types'
...
 
   return (
     <NavigationDotsStyled id="navigation-dots">
-      {navigationConfig.map(({ idScreen }) => (
+      {screenSlidesArray.map((_, index) => (
         <NavigationDotStyled
```
Comment (olenashevchuk) [inline src/app/components/elements/NavigationDots/NavigationDots.js:22]: add key

File: src/app/components/elements/NavigationDots/NavigationDots.styled.js [hunk 4]
```diff
@@ -14,12 +14,14 @@ const NavigationDotStyled = styled.li`
...
-    background-color: var(--booking-module-neutral);
-  }
+  ${(props) =>
+    props?.active &&
+    css`
+      height: 8px;
+      width: 32px;
+      border-radius: 8px;
+      background-color: var(--booking-module-neutral);
+    `}
```
Comment (olenashevchuk) [inline src/app/components/elements/NavigationDots/NavigationDots.styled.js:24]: there is excessive value

File: src/app/components/elements/NavigationDots/NavigationDots.styled.js [hunk 5]
```diff
@@ -14,12 +14,14 @@ const NavigationDotStyled = styled.li`
...
-    border-radius: 8px;
-    background-color: var(--booking-module-neutral);
-  }
+  ${(props) =>
+    props?.active &&
+    css`
+      height: 8px;
+      width: 32px;
+      border-radius: 8px;
+      background-color: var(--booking-module-neutral);
```
Comment (olenashevchuk) [inline src/app/components/elements/NavigationDots/NavigationDots.styled.js:23]: use mediator

File: src/app/hooks/ScreenSlider/useScreenSlider.js [hunk 6]
```diff
@@ -27,10 +27,13 @@ const useScreenSlider = () => {
...
-    ?.querySelector('.screen-slider-wrapper')
-    ?.querySelectorAll('.slick-slide')?.length
-  // get icon for prev button
+  useEffect(() => {
+    const screenSlides = document
+      ?.querySelector('.screen-slider-wrapper')
+      ?.querySelectorAll('.slick-slide')?.length
+    setNumberScreenSlide(screenSlides)
+  }, [])
+
```
Comment (olenashevchuk) [inline src/app/hooks/ScreenSlider/useScreenSlider.js:36]: overcomplexity

File: src/app/hooks/ScreenSlider/useScreenSlider.js [hunk 7]
```diff
@@ -42,16 +45,27 @@ const useScreenSlider = () => {
 
   // [HELPER FUNCTIONS]
   // function that calls the `slickNext` method of Slider
-  const handleNextScreenSlide = () => screenSliderRef?.current?.slickNext()
+  const handleNextScreenSlide = () => {
```
Comment (olenashevchuk) [inline src/app/hooks/ScreenSlider/useScreenSlider.js:48]: not according to standarts, why do you change this code?

File: src/app/hooks/ScreenSlider/useScreenSlider.js [hunk 8]
```diff
@@ -42,16 +45,27 @@ const useScreenSlider = () => {
...
+    screenSliderRef?.current?.slickNext()
+  }
+
   // function that calls the `slickPrev` method of Slider
-  const handlePrevScreenSlide = () => screenSliderRef?.current?.slickPrev()
+  const handlePrevScreenSlide = () => {
```
Comment (olenashevchuk) [inline src/app/hooks/ScreenSlider/useScreenSlider.js:53]: same

File: src/app/hooks/ScreenSlider/useScreenSlider.js [hunk 9]
```diff
@@ -42,16 +45,27 @@ const useScreenSlider = () => {
...
+    screenSliderRef?.current?.slickPrev()
+  }
+
+  const handleInnerScreenSlide = (index) => {
```
Comment (olenashevchuk) [inline src/app/hooks/ScreenSlider/useScreenSlider.js:57]: same

Comment (olenashevchuk) [reply src/app/hooks/ScreenSlider/useScreenSlider.js:57]: not clearly function's name

File: src/app/components/elements/NavigationDots/NavigationDots.styled.js [hunk 10]
```diff
@@ -14,12 +14,14 @@ const NavigationDotStyled = styled.li`
...
-  &.active {
-    height: 8px;
-    width: 32px;
-    border-radius: 8px;
-    background-color: var(--booking-module-neutral);
-  }
+  ${(props) =>
+    props?.active &&
+    css`
+      height: 8px;
```
Comment (olenashevchuk) [inline src/app/components/elements/NavigationDots/NavigationDots.styled.js:20]: useless code

**Review by olenashevchuk — CHANGES_REQUESTED** [review]

File: src/app/components/elements/NavigationDots/NavigationDots.styled.js [hunk 11]
```diff
@@ -14,14 +20,8 @@ const NavigationDotStyled = styled.li`
...
-  ${(props) =>
-    props?.active &&
-    css`
-      height: 8px;
-      width: 32px;
-      border-radius: 8px;
-      background-color: var(--booking-module-neutral);
-    `}
+
+  ${(props) => props?.active && isActiveDot}
```
Comment (olenashevchuk) [inline src/app/components/elements/NavigationDots/NavigationDots.styled.js:24]: use just conditions in styled.li

File: src/app/hooks/ScreenSlider/useScreenSlider.js [hunk 12]
```diff
@@ -45,27 +45,21 @@ const useScreenSlider = () => {
...
+  const handleNextScreenSlide = () => screenSliderRef?.current?.slickNext()
 
   // function that calls the `slickPrev` method of Slider
-  const handlePrevScreenSlide = () => {
-    screenSliderRef?.current?.slickPrev()
-  }
+  const handlePrevScreenSlide = () => screenSliderRef?.current?.slickPrev()
 
-  const handleInnerScreenSlide = (index) => {
+  const handleDifferentScreenSlide = (index) =>
```
Comment (olenashevchuk) [inline src/app/hooks/ScreenSlider/useScreenSlider.js:53]: not a suitable name. think about what this function does

**Review by olenashevchuk — APPROVED** [review]

**Pull request merged**